### DISCUSSION TITLE
Monogame Penumbra Package - Having Background well lit while having tiles and player be affected by light and shadows

### DISCUSSION THREAD

**Discussion by J-i-m-y** [MonoGame/MonoGame#8270] [Graphics]
I'm currently working on a game project and using a lighting library called Penumbra. However, I'm having trouble with it and I was wondering if you who know this library could help me with an issue. I want to have a lit background while also having things like tiles and players be affected by the light. Do you have any suggestions on how I can achieve this? If you could provide me with an article or any helpful resources, that would be great. Thank you!

**Comment by BlizzCrafter** [comment]
You just need to render everything in the right order (look at the [Readme.md](https://github.com/discosultan/penumbra/blob/master/README.md) file of the Penumbra project).

Then you need to choose the right shadowing schemes for your llights. In your case it is probably the `ShadowType.Illuminated` one.

Also take a look at the [Samples](https://github.com/discosultan/penumbra/tree/master/Samples) folder.

**Comment by BlizzCrafter** [comment]
I saw your message via email notification. Nice to hear that your problem is solved :)

**Comment by J-i-m-y** [comment]
I thought I had solved the problem, but it turns out that I haven't. I want to ensure that the background is never affected by changes in light or darkness. I've tried changing the sorting type and rendering it correctly, but I haven't had any success. I've also searched through the documentation, but I haven't found anything for my issue.

**Reply by BlizzCrafter** [reply]
Try the code below to see if it works.

```cs
protected override void Draw(GameTime gameTime)
{
 // Draw your background here !!
  
  penumbra.BeginDraw();
  GraphicsDevice.Clear(Color.CornflowerBlue);
  // Rest of the drawing calls to be affected by Penumbra ...
}
```

**Reply by J-i-m-y** [reply]
Thank you for providing this, but every time I call something before the 'penumbra.BeginDraw();' its never visible

**Reply by BlizzCrafter** [reply]
Ah, damn. You need to draw your background here (even if it sounds unlogical).

```cs
protected override void Draw(GameTime gameTime)
{  
  penumbra.BeginDraw();

  GraphicsDevice.Clear(Color.CornflowerBlue);
  // Affected by Penumbra

  penumbra.Draw(gameTime);
  // Draw your background here !!
}
```

Edit: This hides the penumbra stuff if your background has no transparent areas. You probably need to setup a render target of your background before drawing the penumbra stuff.

**Reply by J-i-m-y** [reply]
Figures, I thought about using a render target, I'll give that a try.

**Reply by BlizzCrafter** [reply]
Well, I thought about your request a second time and I realised that having a lit background while still having light and shadow applied doesn't really make any logical sense :)

Because "shadow" means that some areas of the screen (background) NEED to be darkend... this is the logical definition of a shadow. So you can't have a background to be uneffected of that. It's kinda the same for lights with brightness.

But what you can do is to adjust the ratio of shadow and light affectedness of your screen (background) with the built in property "AmbientColor" of the penumbra component.

Look at these pictures:

`AmbientColor = new Color(new Vector3(0.1f))`
![Platformer2D_BTzuOOps2k](https://github.com/MonoGame/MonoGame/assets/7989230/6273bd27-ed7c-489b-902f-72e9c2ec8904)

`AmbientColor = new Color(new Vector3(0.8f))`
![Platformer2D_AmbientLess](https://github.com/MonoGame/MonoGame/assets/7989230/09f3b5c8-aab6-4727-97a1-c0d6849b7f1c)

You could set this property when creating the penumbra component in your game class like this:

```c#
penumbra = new PenumbraComponent(this)
{
    AmbientColor = new Color(new Vector3(0.8f))
};
```

**Reply by J-i-m-y** [reply]
Well Damn, Thank you
